web/packages/design/src/SlideTabs/SlideTabs.tsx

```diff
@@ -16,40 +16,44 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import React, { useEffect, useState } from 'react';
+import React from 'react';
 import styled from 'styled-components';
 
-function SlideTabs({
+import { Indicator, Text } from '..';
+
+export function SlideTabs({
   appearance = 'square',
-  initialSelected = 0,
+  activeIndex = 0,
   name = 'slide-tab',
   onChange,
   size = 'xlarge',
   tabs,
+  isProcessing = false,
 }: props) {
-  const [activeIndex, setActiveIndex] = useState(initialSelected);
-
-  useEffect(() => {
-    onChange(activeIndex);
-  }, [activeIndex]);
-
   return (
     <Wrapper>
       <TabNav role="tablist" appearance={appearance} size={size}>
-        {tabs.map((tabData, tabIndex) => {
-          const tabDataType = typeof tabData === 'string';
-          const tabName = tabDataType ? tabData : tabData.name;
-          const tabContent = tabDataType ? tabData : tabData.component;
+        {tabs.map((tabName, tabIndex) => {
+          const selected = tabIndex === activeIndex;
           return (
             <TabLabel
               role="tab"
               htmlFor={`${name}-${tabName}`}
-              onClick={() => setActiveIndex(tabIndex)}
+              onClick={e => {
+                e.preventDefault();
```

Comment:
Is this `e.preventDefault()` necessary? From what I see, this is just a `<label>`, so there's no extra behavior on click. Pointer events are turned off when it's processing too.

Reply:
yeah i was confused by it to, but without the `e.preventDefault`, onChange gets called twice for some reason